pkg/collector/corechecks/cluster/ksm/kubernetes_state.go

```diff
@@ -568,6 +614,39 @@ func manageResourcesReplacement(c *apiserver.APIClient, factories []customresour
 
 // Run runs the KSM check
 func (k *KSMCheck) Run() error {
+	var client *apiserver.APIClient
+
+	// Perform resource discovery if it hasn't successfully completed yet
+	if k.resourceDiscoveryInfo == nil {
+		// Acquire the API Server client, retrying with exponential backoff
+		apiCtx, apiCancel := context.WithTimeout(context.Background(), maximumWaitForAPIServer)
+		defer apiCancel()
+
+		var err error
+		// WaitForAPIClient handles its own retries
+		client, err = apiserver.WaitForAPIClient(apiCtx)
```

Comment:
Not sure you need to wait, you can just call `GetAPIClient`. If you don't get it it will be retried anyway next check run.